pkg/scheduler/algorithm/predicates/utils.go

```diff
@@ -147,3 +148,193 @@ func isCSIMigrationOn(csiNode *storagev1beta1.CSINode, pluginName string) bool {
 
 	return mpaSet.Has(pluginName)
 }
+
+// utilities for building pod/node objects using a "chained" manner
```

Comment:
I am not sure what "chained" means here, can you give an example?

Reply:
Also, where are those wrappers being used? I couldn't find any reference to them in your PRs (I think).

Reply:
Maybe "pipelined" is a better name? It in particular means a pattern that after calling foo(), the caller passed in is still returned so that we can streamline multiple callings with less code, like obj.foo().bar().baz()... In Kubernetes, we have similar pattern like clisentset().Namespace().Name()....

The wrappers are now only used in this feature's test cases. But I believe it can be reused to refactor other test cases.

Reply:
> The wrappers are now only used in this feature's test cases. But I believe it can be reused to refactor other test cases.

If they are to aid testing, can we move them to test-only util files? Also, it seems that the logic here is the same as the one in  "pkg/scheduler/util/utils.go" (one is exported, the other is not), can we please remove this redundancy?

Reply:
> can we please remove this redundancy?

Oh my bad... Apologize for the confusion. They should reside in pkg/scheduler/util/utils.go and be public.

(I guess it's due to my sloppy rebase)

> If they are to aid testing, can we move them to test-only util files?

How about creating a pkg/scheduler/util/testing.go?

Reply:
Does it need to be in a generic directory like pkg/scheduler/util? for example we have a test_util.go file under pkg/scheduler/algorithm/priorities/test_util.go, it might be better to do the same for this: place it in a file named test_util.go under the package path where it is used.

Reply:
For most cases that pattern works. But in this (particular) case, the helper functions are used by multiple packages<sup>1</sup>, so pkg/scheduler/util seems to be a good place. (also can avoid cyclic dependency issue)

<sup>1</sup> Precisely speaking, it's only used in one package, but will be used in generic_scheduler as well as priority package.

Reply:
ok, in that case, I think the best place for it is under the scheduler/testing, name the file utils.go and the package name is testing, how does that sound?

Otherwise, this looks good to me, and this one is just blocked on the API PR.

Reply:
Sounds good. Updated.